src/common.js

```diff
@@ -274,7 +283,24 @@ function wrap(cmd, fn, options) {
         });
         if (!config.noglob && options && typeof options.idx === 'number')
           args = args.slice(0, options.idx).concat(expand(args.slice(options.idx)));
-        retValue = fn.apply(this, args);
+        if (config.async || callback) {
+          process.nextTick(function() {
+            var r = fn.apply(this, args);
+
+            // Convert return value to an object that makes sense
+            if (typeof r === 'boolean')
+              r = {stdout: '', stderr: '', code: (r ? 0 : 1)};
+            else if (typeof r === 'string')
+              r = {stdout: r, stderr: (state.error || ''), code: (state.error ? 1 : 0)};
+            else if (typeof r === 'undefined')
+              r = {stdout: '', stderr: (state.error || ''), code: (state.error ? 1 : 0)};
+
+            if (callback)
+              callback.call(this, r.code, r.stdout, r.stderr);
```

Comment:
The issue is that this is inconsistent with `exec()`, which already takes a callback of this form.

With these parameters, the user can construct the equivalent `ShellString`.